source/test/F0.Minesweeper.Components.Tests/MinefieldTests.cs

```diff
@@ -74,13 +92,146 @@ public void Rendering_OneCellShown_MarkupIsCorrect()
 			// Arrange
 			const string expectedMarkup = "<div><h3>Minefield</h3><table><tr><td><div diff:ignore /></td></tr></table></div>";
 
-			ComponentParameter parameter = ComponentParameterFactory.Parameter(nameof(Minefield.Size), new MinefieldSize(1, 1));
+			var options = new MinefieldOptions(1, 1, 2, MinefieldFirstUncoverBehavior.MayYieldMine);
+
+			minefieldFactoryMock.Setup(factory => factory.Create(options)).Returns(minefieldMock.Object);
+
+			ComponentParameter parameter = ComponentParameterFactory.Parameter(nameof(Minefield.Options), options);
 
 			// Act
 			IRenderedComponent<Minefield> componentUnderTest = RenderComponent<Minefield>(parameter);
 
 			// Assert
 			componentUnderTest.MarkupMatches(expectedMarkup);
 		}
+
+		[Fact]
+		public void OnCellUncoveredAsync_MinefieldWasNotCreated_Throws()
+		{
+			// Arrange
+			var options = new MinefieldOptions(1, 1, 2, MinefieldFirstUncoverBehavior.MayYieldMine);
+
+			minefieldFactoryMock.Setup(factory => factory.Create(options)).Returns<IMinefield>(null);
+
+			ComponentParameter parameter = ComponentParameterFactory.Parameter(nameof(Minefield.Options), options);
+
+			IRenderedComponent<Minefield> componentUnderTest = RenderComponent<Minefield>(parameter);
+
+			// Act && Assert
+			Action actionToTest = () => componentUnderTest.Find("button").Click();
+			actionToTest.Should().Throw<InvalidOperationException>();
+		}
+
+		[Theory]
+		[MemberData(nameof(GetReportVariations))]
+		private void OnCellUncoveredAsync_ReportVariations_DoesNotThrow(GameUpdateReportForTests report)
+		{
+			// Arrange
+			var options = new MinefieldOptions(1, 1, 2, MinefieldFirstUncoverBehavior.MayYieldMine);
+
+			minefieldFactoryMock.Setup(factory => factory.Create(options)).Returns(minefieldMock.Object);
+			minefieldMock.Setup(field => field.Uncover(It.Is<Location>(s => s == new Location(0, 0)))).Returns(report);
+
+			ComponentParameter parameter = ComponentParameterFactory.Parameter(nameof(Minefield.Options), options);
+
+			IRenderedComponent<Minefield> componentUnderTest = RenderComponent<Minefield>(parameter);
+
+			// Act && Assert
+			Action actionToTest = () => componentUnderTest.Find("button").Click();
+			actionToTest.Should().NotThrow();
+		}
+
+		private static IEnumerable<object[]> GetReportVariations()
```

Comment:
We could use strongly typed `TheoryData` rather than `object[]`: [Creating strongly typed xUnit theory test data with TheoryData](https://andrewlock.net/creating-strongly-typed-xunit-theory-test-data-with-theorydata/).
And in combination with Target-Typed new expressions, this shouldn't be too much additional code.

Reply:
Been there, done that :)